.github/workflows/build-system-test.yml

```diff
@@ -0,0 +1,148 @@
+# Description: this workflow runs build mega-apps that have all the connected components.
+
+name: Build System Test Canary
+
+permissions:
+  id-token: write # This is required for aws-actions/configure-aws-credentials
```

Comment:
L5 and L6 is the only difference from https://github.com/aws-amplify/amplify-ui/pull/3703